code/modules/xenoarchaeology/tools/console/xenoarchaeology_console.dm

```diff
@@ -0,0 +1,261 @@
+			current_pack.contains = list()
+			current_pack?.current_supply = max(1, current_pack.current_supply) //Don't worry about it :^)
+			current_pack.cost += seller?.get_price(locate(params["item_id"]))
+			current_pack.contains += seller?.buy_stock(locate(params["item_id"]))
+			current_order = new /datum/supply_order(current_pack, name, rank, ckey, "Research Material Requisition", D)
+			current_order.generateRequisition(get_turf(src))
+			SSsupply.shoppinglist += current_order
+			console_orders[current_order] = current_pack
+		//Radio shit
+		if("toggle_purchase_audio")
+			radio_purchase_notice = !radio_purchase_notice
+		if("toggle_solved_audio")
+			radio_solved_notice = !radio_solved_notice
+
+	ui_update()
+
+/obj/machinery/computer/xenoarchaeology_console/proc/check_sold(datum/source, atom/movable/AM, sold)
+	SIGNAL_HANDLER
+
+	var/obj/item/sticker/xenoartifact_label/L = AM
+	if(!istype(L))
+		return
+	var/atom/artifact = L.loc
+	var/datum/component/xenoartifact/X = artifact?.GetComponent(/datum/component/xenoartifact)
+	if(!X || !artifact)
+		return
+	//Grab values to calculate success
+	var/score = 0
+	var/max_score = 0
+	var/bonus = 0
+	var/max_bonus = 0
+	var/attempted_bonus = FALSE
+	var/list/traits_by_type = list()
+	for(var/i in X.artifact_traits) //By priority
+		for(var/datum/xenoartifact_trait/T in X.artifact_traits[i]) //By trait in priorty
+			traits_by_type += list(T.type)
+			if(T.contribute_calibration)
+				max_score += 1
+			else
+				max_bonus += 1
+	for(var/datum/xenoartifact_trait/T as() in L.traits)
+		if((T in traits_by_type))
+			if(initial(T.contribute_calibration))
+				score += 1
+			else
+				bonus += 1
+				attempted_bonus = TRUE
+		else
+			if(initial(T.contribute_calibration))
+				score -= 1
+			else
+				bonus -= 1
+				attempted_bonus = TRUE
+	//Calculate success rate
+	var/success_rate = score / (max_score || 1)
+	var/bonus_rate = max(1, 2*(bonus/(max_bonus||1)))
+	//Rewards
+		//Research Points
+	var/rnd_reward = max(0, (artifact.custom_price*X.artifact_type.rnd_rate)*success_rate) * bonus_rate
+		//Discovery Points
+	var/dp_reward = max(0, (artifact.custom_price*X.artifact_type.dp_rate)*success_rate) * bonus_rate
+		//Money //TODO: Check if this is sufficient - Racc : PLAYTEST
+	var/monetary_reward = ((artifact.custom_price * success_rate * 1.5)^1.1) * (success_rate >= 0.5 ? 1 : 0) * bonus_rate
+	//Alloctae
+	if(main_console)
+		linked_techweb?.add_point_type(TECHWEB_POINT_TYPE_GENERIC, rnd_reward)
+		linked_techweb?.add_point_type(TECHWEB_POINT_TYPE_DISCOVERY, dp_reward)
+		budget.adjust_money(monetary_reward)
+	//Announce victory or fuck up
+	var/success_type
+	switch(success_rate)
+		if(0.3 to 0.69)
+			success_type = "sufficient research"
+		if(0.7 to 0.89)
+			success_type = "admirable research"
+		if(0.9 to INFINITY)
+			success_type = "incredible discovery"
+		else
+			success_type = prob(1) ? "scientific failure." : "who let the clown in?"
```

Comment:
clown message has 99% chance to appear. did you mistake this secret funness?

Reply:
Oops